tests/workflows/test_consistency.py

```diff
@@ -0,0 +1,235 @@
+
+    # subprocess.call(["docker", "run", "--volume", f"{tmp_path}:{tmp_path}", "--rm", "halfpipe/halfpipe:1.2.2 "])
+
+    index = BIDSIndex()
+    index.put(tmp_path / "derivatives")
+    spec_file = tmp_path / "spec.json"
+
+    for sub in dataset.subject_ids:
+        paths_to_zip = []
+        for test_setting in settings_list:
+            name = test_setting.name
+
+            # Search for files we want to save in each setting
+            timeseries = index.get(sub=sub, feature=f"{name}CorrMatrix", suffix="timeseries", task="rest", extension=".tsv")
+            corr = index.get(sub=sub, feature=f"{name}CorrMatrix", desc="correlation", suffix="matrix")
+            dual_reg = index.get(sub=sub, feature=f"{name}DualReg", suffix="statmap", stat="z", component="8")
+            falff = index.get(sub=sub, feature=f"{name}FALFF", suffix="falff", extension=".nii.gz")
+            alff = index.get(sub=sub, feature=f"{name}FALFF", suffix="alff", extension=".nii.gz")
+            reho = index.get(sub=sub, feature=f"{name}ReHo", suffix="reho", extension=".nii.gz")
+            pcc = index.get(sub=sub, feature=f"{name}SeedCorr", suffix="statmap", stat="z")
+
+            # Check that halfpipe processing generated those files
+            for feature_name, feature_path in [
+                ("timeseries", timeseries),
+                ("correlation matrix", corr),
+                ("dual regression", dual_reg),
+                ("falff", falff),
+                ("alff", alff),
+                ("reho", reho),
+                ("pcc", pcc),
+            ]:
+                assert feature_path is not None and len(feature_path) == 1, f"Missing path for {name} {feature_name}"
+
+            paths_to_zip.extend(
+                [
+                    list(timeseries or [])[0],
+                    list(corr or [])[0],
+                    list(dual_reg or [])[0],
+                    list(falff or [])[0],
+                    list(alff or [])[0],
+                    list(reho or [])[0],
+                    list(pcc or [])[0],
+                ]
+            )
+
+        # Search for files we want to save at the subject level
+        tsnr_fmriprep = index.get(sub=sub, suffix="tsnr", datatype="func")
+        paths_to_zip.extend([list(tsnr_fmriprep or [])[0], spec_file])
+
+        # Create the zip file in the specified output directory
+        timestamp = datetime.now().strftime("%Y%m%d-%H%M%S")
+        zip_filename = f"{dataset.openneuro_id}_sub-{sub}_time-{timestamp}.zip"
+        zip_filepath = tmp_path / zip_filename
+        with zipfile.ZipFile(zip_filepath, "w") as zipf:
+            for file in paths_to_zip:
+                # Ensure file is a Path instance and convert it to string if needed
+                if isinstance(file, Path):
+                    zipf.write(str(file), arcname=str(file.relative_to(tmp_path)))
+                else:
+                    raise TypeError(f"Unexpected type for file: {type(file)}")
```

Comment:
```suggestion
                if not isinstance(file, Path):
                    raise TypeError(f"Unexpected type for file: {type(file)}")
                zipf.write(str(file), arcname=str(file.relative_to(tmp_path)))
```
I would suggest using the guard clause pattern here.